src/test/java/de/tum/in/www1/artemis/LearningGoalIntegrationTest.java

```diff
@@ -451,15 +452,22 @@ public void getLearningGoalCourseProgressTeamsTest_asInstructorOne() throws Exce
 
         assertThat(courseLearningGoalProgress.totalPointsAchievableByStudentsInLearningGoal).isEqualTo(30.0);
         assertThat(courseLearningGoalProgress.averagePointsAchievedByStudentInLearningGoal).isEqualTo(3.0);
-        CourseLearningGoalProgress.CourseLectureUnitProgress progressInTeamExercise = courseLearningGoalProgress.progressInLectureUnits.stream()
-                .filter(courseLectureUnitProgress -> courseLectureUnitProgress.lectureUnitId.equals(idOfExerciseUnitTeamTextOfLectureOne)).findFirst().get();
-        assertThat(progressInTeamExercise.participationRate).isEqualTo(80.0);
-        assertThat(progressInTeamExercise.noOfParticipants).isEqualTo(4);
-        assertThat(progressInTeamExercise.averageScoreAchievedByStudentInLectureUnit).isEqualTo(30.0);
+
+        boolean foundProgressWithCorrectNumbers = false;
+        for (CourseLearningGoalProgress.CourseLectureUnitProgress courseLectureUnitProgress : courseLearningGoalProgress.progressInLectureUnits) {
+            if (courseLectureUnitProgress.participationRate.equals(80.0) && courseLectureUnitProgress.noOfParticipants.equals(4)
+                    && courseLectureUnitProgress.averageScoreAchievedByStudentInLectureUnit.equals(30.0)) {
+                foundProgressWithCorrectNumbers = true;
+                break;
+            }
+        }
+
+        assertThat(foundProgressWithCorrectNumbers).isTrue();
     }
 
     @Test
     @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    @RepeatedTest(100)
```

Comment:
please remove this again after it works properly

Reply:
Done in fbc81e7baa16cd07ac9619377b780a39545b7614